poutyne/framework/model.py

```diff
@@ -685,6 +686,36 @@ def _transfer_optimizer_state_to_right_device(self):
                         if torch.is_tensor(v) and p.device != v.device:
                             v.data = v.data.to(p.device)
 
+    def _get_named_optimizer_attrs(self):
+        param_to_name = {param: name for name, param in self.model.named_parameters()}
+
+        param_name_groups = []
+        for group in self.optimizer.param_groups:
+            param_name_groups.append([param_to_name[param] for param in group['params']])
+
+        named_state = {param_to_name[param]: state for param, state in self.optimizer.state.items()}
+
+        return param_name_groups, named_state
+
+    def _set_named_optimizer_attrs(self, param_name_groups, named_state):
+        name_to_param = dict(self.model.named_parameters())
+
+        for param_name_group, optim_group in zip(param_name_groups, self.optimizer.param_groups):
+            optim_group['params'] = [
+                name_to_param[param_name] if optim_param is not name_to_param[param_name] else optim_param
+                for param_name, optim_param in zip(param_name_group, optim_group['params'])
+            ]
+
+        self.optimizer.state = {name_to_param[name]: state for name, state in named_state}
+
+    @contextlib.contextmanager
+    def _update_optim_device_context_manager(self):
```

Comment:
I think the name should not include _context_manager. 

As a comparison, when using open as a context manager, the function is not open_context_manager.